pandas/core/arrays/base.py

```diff
@@ -657,6 +691,25 @@ def searchsorted(self, value, side="left", sorter=None):
         arr = self.astype(object)
         return arr.searchsorted(value, side=side, sorter=sorter)
 
+    def equals(self, other: ABCExtensionArray) -> bool:
+        """
+        Return if another array is equivalent to this array.
+
+        Parameters
+        ----------
+        other: ExtensionArray
+            Array to compare to this Array.
+
+        Returns
+        -------
+        boolean
+            Whether the arrays are equivalent.
+
+        """
+        return isinstance(other, self.__class__) and (
```

Comment:
Perhaps `type(self) == type(other)`? It's not clear to me whether a subclass should compare `.equals` to a parent class. Thoughts @jreback?

Reply:
Will want some length checking here too.

Reply:
Added `type(self) == type(other)` and length checks.